lib/Backend/Commands.hs

```diff
@@ -100,6 +104,28 @@ createCmd (CreateOptions entryPath _edit force tags fields privateFields) = do
   void $ writeSecret entry
   pure $ CRSuccess entry
 
+  where
+    -- | When attempting to create an entry at, e.g., @/a/b/c@, the directories
+    -- @/a@ and @/a/b@ will be implicitly created.
+    --
+    -- This function checks whether those paths are already
+    -- occupied by existing entries and, if so, fails.
+    --
+    -- Note: the root path @/@ cannot possibly be occupied by an entry,
+    -- therefore we skip the check for that path.
+    checkForEntriesInEntryPath :: EntryPath -> Sem r ()
+    checkForEntriesInEntryPath entryPath = do
+      let parentDirsExceptRoot = entryPath
+            & Path.entryPathParentDirs
+            & NE.toList
+            -- Ignore paths that cannot possibly point to an entry
+            <&> Path.pathAsEntryPath
+            & rights
```

Comment:
`filterM` wouldn't fit here, because we also need to cast from `Path` to `EntryPath`.

Though, `mapMaybe` + `rightToMaybe` would work. But I don't think I would personally prefer that over `fmap` + `rights`, in this instance...